sdk/core/azure-core/inc/azure/core/http/curl/curl connection_pool.hpp

```diff
@@ -27,6 +27,84 @@ namespace Azure { namespace Core { namespace Test {
 
 namespace Azure { namespace Core { namespace Http {
 
+  /**
+   * @brief The available options to set libcurl ssl options.
+   *
+   * @remark The SDK will map the enum option to libcurl's specific option. See more info here:
+   * https://curl.haxx.se/libcurl/c/CURLOPT_SSL_OPTIONS.html
+   *
+   */
+  struct CurlTransportSSLOptions
+  {
+    bool AllowBeast = false;
+    bool NoRevoke = false;
+    /*
+    // Requires libcurl version >= 7.68
+    bool NoPartialchain = false;
+    // Requires libcurl version >= 7.70
+    bool RevokeBestEffort = false;
+    // Requires libcurl version >= 7.71
+    bool NativeCa = false;
+    */
+  };
+
+  /**
+   * @brief Set the curl connection options like a proxy and CA path.
+   *
+   */
+  struct CurlTransportOptions
+  {
+    /**
+     * @brief The string for the proxy is passed directly to the libcurl handle without any parsing
+     *
+     * @remark No validation for the string is done by the Azure SDK. More about this option:
+     * https://curl.haxx.se/libcurl/c/CURLOPT_PROXY.html.
+     *
+     * @remark The default value is an empty string (no proxy).
+     *
+     */
+    std::string Proxy;
+    /**
+     * @brief The string for the certificate authenticator is sent to libcurl handle directly.
+     *
+     * @remark The Azure SDK will not check if the path is valid or not.
+     *
+     * @remark The default is the built-in system specific path. More about this option:
+     * https://curl.haxx.se/libcurl/c/CURLOPT_CAINFO.html
+     *
+     */
+    std::string CAInfo;
+    /**
+     * @brief All HTTP requests will keep the connection channel open to the service.
+     *
+     * @remark The channel might be closed by the server if the server response has an error code.
+     * A connection won't be re-used if it is abandoned in the middle of an operation.
+     * operation.
+     *
+     * @remark This option is managed directly by the Azure SDK. No option is set for the curl
+     * handle. It is `true` by default.
+     */
+    bool HttpKeepAlive = true;
+    /**
+     * @brief This option determines whether curl verifies the authenticity of the peer's
+     * certificate.
+     *
+     * @remark The default value is `true`. More about this option:
+     * https://curl.haxx.se/libcurl/c/CURLOPT_SSL_VERIFYPEER.html
+     *
+     */
+    bool SSLVerifyPeer = true;
+
+    /**
+     * @brief Define the SSL options for the libcurl handle.
+     *
+     * @remark See more info here: https://curl.haxx.se/libcurl/c/CURLOPT_SSL_OPTIONS.html.
+     * The default option is all options `false`.
+     *
+     */
+    CurlTransportSSLOptions SSLOptions;
```

Comment:
Why separate out the SSLOptions from other Curl options such as SSLVerifyPeer?  I'd prefer all the options to be flat on the options struct.